apps/src/gamelab/animationListModule.js

```diff
@@ -0,0 +1,557 @@
+ * @param {!AnimationKey} key
+ * @param {string} [sourceUrl]
+ * @param {function} [callback]
+ */
+function loadAnimationFromSource(key, sourceUrl, callback) {
+  // Figure out URL from animation key
+  // TODO: Take version ID into account here...
+  sourceUrl = sourceUrl || animationsApi.basePath(key) + '.png';
+  callback = callback || function () {};
+  return dispatch => {
+    dispatch({
+      type: START_LOADING_FROM_SOURCE,
+      key: key
+    });
+    fetchUrlAsBlob(sourceUrl, (err, blob) => {
+      if (err) {
+        console.log('Failed to load animation ' + key, err);
+        // Brute-force recovery step: Remove the animation from our redux state;
+        // it looks like it's already gone from the server.
+        dispatch({
+          type: DELETE_ANIMATION,
+          key
+        });
+        return;
+      }
+
+      blobToDataURI(blob, dataURI => {
+        dispatch({
+          type: DONE_LOADING_FROM_SOURCE,
+          key,
+          blob,
+          dataURI
+        });
+        callback();
+      });
+    });
+  };
+}
+
+function fetchUrlAsBlob(url, onComplete) {
+  let xhr = new XMLHttpRequest();
+  xhr.open('GET', url, true);
+  xhr.responseType = 'blob';
+  xhr.onload = e => {
+    if (e.target.status === 200) {
+      onComplete(null, e.target.response);
+    } else {
+      onComplete(new Error(`URL ${url} responded with code ${e.target.status}`));
+    }
+  };
+  xhr.onerror = e => onComplete(new Error(`Error ${e.target.status} occurred while receiving the document.`));
+  xhr.send();
+}
+
+function blobToDataURI(blob, onComplete) {
+  let fileReader = new FileReader();
+  fileReader.onload = e => onComplete(e.target.result);
+  fileReader.readAsDataURL(blob);
+}
+
+/**
+ * Dispatch to save animations to S3.
+ * @param {function} onComplete callback - when all animations are saved
+ * @returns {function()}
+ */
+export function saveAnimations(onComplete) {
+  return (dispatch, getState) => {
+    const state = getState().animationList;
+    const changedAnimationKeys = state.list.filter(key => !state.data[key].saved);
+    Promise.all(changedAnimationKeys.map(key => {
+          return saveAnimation(key, state.data[key])
+              .then(action => { dispatch(action); });
+        }))
+        .then(() => {
+          console.log(`Saved ${changedAnimationKeys.length} animations to S3`); // TODO: Remove
+          onComplete();
+        })
+        .catch(err => {
+          // TODO: What should we really do in this case?
+          console.log('Failed to save animations', err); // TODO: Remove?
+          onComplete();
+        });
+  };
+}
+
+/**
+ *
+ * @param {AnimationKey} animationKey
+ * @param {Object} animationData // TODO: Better type?
+ * @return {Promise} which resolves to a redux action object
+ */
+function saveAnimation(animationKey, animationData) {
+  return new Promise((resolve, reject) => {
+    if (typeof animationData.blob === 'undefined') {
```

Comment:
Having trouble working out where I got the idea that this is recommended, but here's the underlying reason: `typeof x === 'undefined'` will work for undeclared variables, where `x === undefined` will produce a reference error, which is generally only an issue for global scope (hence, always `typeof window === 'undefined'`when trying to be node-friendly).  That may not be an issue in this case. It's also resistant to `undefined` accidentally getting shadowed within a function.
[(source: stackOverflow)](http://stackoverflow.com/a/4725697/5000129)

I can't find this in any of our styleguides though.  Do you prefer the strict equality check?